src/dmd/dcast.d

```diff
@@ -1508,6 +1487,32 @@ Expression castTo(Expression e, Scope* sc, Type t)
             this.t = t;
         }
 
+        // Try casting the alias this member. Return the expression if it succeeds, null otherwise.
```

Comment:
What are tob and t1b ?

Reply:
Am I correct that moving tryAliasThisCast here was just a refactoring, and not part of the fix? If so, refactorings (no matter how tempting) should go in separate PRs.

Reply:
It is a refactoring, but it also adds some code.

Reply:
But you are right, the refactoring does make it difficult to spot the added bits.

Reply:
Yep sorry, I've separated refactoring and fix in two commits. It looks more clear now.